### DISCUSSION TITLE
How do I delete from a state ?

### DISCUSSION THREAD

**Discussion by pzdkn** [writer/writer-framework#326] [Q&A] [answered]
I have a dict stored in a state that I use for a repeater. With a button click, I want to remove elements from this dict. 

It seems I can't do regular, dict operations for it e.g. `del state["repeater_state]["item_1"]` or `state["repeater_state"].pop("item_1") `.  Now reassigning the repeater_state by filtering out "item_1" is not reflected in the repeater frontend.

 Whats the recommended way to do it ?

**Comment by pzdkn** [comment]
I found this:

https://github.com/streamsync-cloud/streamsync/discussions/168

but is there a more elegant solution than reconstructing the dict ? It seems very resource-intense to me.

**Reply by FabienArcellier** [reply, marked as answer]
A more elegant is coming thanks to the work of @mmikita95 

In the next release (the next week), you would be able to just `del state["repeater_state]["item_1"]`. It's already in the version on dev branch. 

https://github.com/streamsync-cloud/streamsync/pull/214

**Reply by FabienArcellier** [reply]
This change should work on the last realease 0.4.0.
